pennylane/tape/tape.py

```diff
@@ -40,29 +46,40 @@ def _err_msg_for_some_meas_not_qwc(measurements):
     )
 
 
-def _validate_computational_basis_sampling(measurements):
+def _validate_computational_basis_sampling(tape):
```

Comment:
I'm a little curious about what forcing the change in this code.  If it's a mid-circuit measurement, shouldn't it be completely independent of what basis we are measuring in at the end?

Reply:
I agree, but without this change, the (old API) L-Kokkos device raises an error
```
pennylane.QuantumFunctionError: Only observables that are qubit-wise commuting Pauli words can be returned on the same wire, some of the following measurements do not commute:
```
The current resolution feels hacky, but I could not find a sleeker way to resolve this.

Reply:
weird but ok 🤔